pkg/virt-controller/watch/vm.go

```diff
@@ -3133,6 +3133,20 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 		return vm, &syncErrorImpl{fmt.Errorf("error encountered while upgrading instancetype.kubevirt.io ControllerRevisions: %v", err), FailedCreateVirtualMachineReason}, nil
 	}
 
+	// eventually, would like the condition to be `== "true"`, but for now we need to support legacy behavior by default
+	if vm.Annotations[virtv1.ImmediateDataVolumeCreation] != "false" {
```

Comment:
not sure if its worth to add this check with != "true" where you do this exact same code in the startVMI code to prevent from this handleDatavolumes from running twice..

Reply:
Are you saying you would prefer:

```golang
if val, ok := vm.Annotations[virtv1.ImmediateDataVolumeCreation]; !ok || val == "true" {
```

that way if the annotation value is "baz" `handleDataVolumes will not run?

Reply:
I question why we need to expose a per-VM option for this at all.  I think we should change this behavior for everyone given the justification you provide in the PR.  If a user wants immediate provisioning they are free to start and stop the VM.

Reply:
Is there a reason we want to keep the legacy behavior as default? If so, I'm ok with starting with the per-VM and eventually change the condition to `== true`.

Reply:
@aglitke @alromeros 

Would like to merge this pr as-is in order to easily support backporting with "legacy" behavior as default while having the option for `kubevirt.io/immediate-data-volume-creation: "false"`.  I think this behavior change is too big to force on existing releases.

I will open a subsequent PR only for main that changes this condition to:  `if vm.Annotations[virtv1.ImmediateDataVolumeCreation] != "true"`

I think the per-VM configuration is helpful for users that may like the "immediate" behavior for whatever reason

Reply:
Makes sense, the PR looks good to me then, let's see if @ShellyKa13 is ok with the changes.